components/web/firefox/patches/webrtc-solaris.patch

```diff
@@ -0,0 +1,19393 @@
+index 1c228d3ef..97c92a312 100644
+--- firefox-60.8.0/media/webrtc/trunk/webrtc/audio/utility/audio_frame_operations_gn/moz.build
++++ firefox-patched/media/webrtc/trunk/webrtc/audio/utility/audio_frame_operations_gn/moz.build
+@@ -104,6 +104,13 @@ if CONFIG["OS_TARGET"] == "OpenBSD":
+     DEFINES["WEBRTC_POSIX"] = True
+     DEFINES["_FILE_OFFSET_BITS"] = "64"
+ 
++if CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["USE_X11"] = "1"
++    DEFINES["WEBRTC_POSIX"] = True
++    DEFINES["WEBRTC_SOLARIS"] = True
++    DEFINES["_FILE_OFFSET_BITS"] = "64"
++
+ if CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["CERT_CHAIN_PARA_HAS_EXTRA_FIELDS"] = True
+@@ -153,6 +160,10 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+ 
+     DEFINES["_FORTIFY_SOURCE"] = "2"
+ 
++if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["_FORTIFY_SOURCE"] = "2"
++
+ if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
+diff --git firefox-60.8.0/media/webrtc/trunk/webrtc/base/gtest_prod_gn/moz.build firefox-patched/media/webrtc/trunk/webrtc/base/gtest_prod_gn/moz.build
+index 3cfd5128a..2c9aa225e 100644
+--- firefox-60.8.0/media/webrtc/trunk/webrtc/base/gtest_prod_gn/moz.build
++++ firefox-patched/media/webrtc/trunk/webrtc/base/gtest_prod_gn/moz.build
+@@ -96,6 +96,13 @@ if CONFIG["OS_TARGET"] == "OpenBSD":
+     DEFINES["WEBRTC_POSIX"] = True
+     DEFINES["_FILE_OFFSET_BITS"] = "64"
+ 
++if CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["USE_X11"] = "1"
++    DEFINES["WEBRTC_POSIX"] = True
++    DEFINES["WEBRTC_SOLARIS"] = True
++    DEFINES["_FILE_OFFSET_BITS"] = "64"
++
+ if CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["CERT_CHAIN_PARA_HAS_EXTRA_FIELDS"] = True
+@@ -145,6 +152,10 @@ if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "OpenBSD":
+ 
+     DEFINES["_FORTIFY_SOURCE"] = "2"
+ 
++if not CONFIG["MOZ_DEBUG"] and CONFIG["OS_TARGET"] == "SunOS":
++
++    DEFINES["_FORTIFY_SOURCE"] = "2"
++
+ if CONFIG["MOZ_DEBUG"] == "1" and CONFIG["OS_TARGET"] == "WINNT":
+ 
+     DEFINES["_HAS_ITERATOR_DEBUGGING"] = "0"
+diff --git firefox-60.8.0/media/webrtc/trunk/webrtc/base/platform_thread.cc firefox-patched/media/webrtc/trunk/webrtc/base/platform_thread.cc
+index e72640613..a5980da13 100644
+--- firefox-60.8.0/media/webrtc/trunk/webrtc/base/platform_thread.cc
++++ firefox-patched/media/webrtc/trunk/webrtc/base/platform_thread.cc
+@@ -21,6 +21,8 @@
+ #include <lwp.h>
+ #elif defined(__FreeBSD__)
+ #include <pthread_np.h>
++#elif defined(__sun)
++#include <thread.h>
+ #endif
+ 
+ namespace rtc {
+@@ -53,6 +55,8 @@ PlatformThreadId CurrentThreadId() {
+   ret = reinterpret_cast<uintptr_t> (pthread_self());
+ #elif defined(__FreeBSD__)
+   ret = pthread_getthreadid_np();
++#elif defined(__sun)
++  ret = thr_self();
```

Comment:
Should it be possible to use pthread_self() as others do?

Reply:
the code underneath it runs afoul of signed/unsigned int conversion, and according to this, pthread is implemented in terms of libthread: https://docs.oracle.com/cd/E26502_01/html/E35303/compile-94611.html#compile-2
```
When using -mt, the Oracle Solaris threads APIs will be linked automatically. Always use the -mt option instead of listing -lthread explicitly. To use Pthreads, specify the -mt option and -lpthread option on the link command line. The libpthread library provides an interface to libthread, so you still need libthread when using Pthreads.```

Reply:
I understand, but I'd prefered to be 'as others', one syscall is not such an improvement, but code compatibility is.